test/k4FWCoreTest/src/components/k4FWCoreTest_AlgorithmWithTFile.h

```diff
@@ -47,9 +49,7 @@ class k4FWCoreTest_AlgorithmWithTFile : public GaudiAlgorithm {
   /// Handle for the genvertices to be written
   DataHandle<edm4hep::SimTrackerHitCollection> m_simTrackerHitHandle{"SimTrackerHit", Gaudi::DataHandle::Writer, this};
 
-  DataHandle<float>              m_singleFloatHandle{"SingleFloat", Gaudi::DataHandle::Writer, this};
-  DataHandle<std::vector<float>> m_vectorFloatHandle{"VectorFloat", Gaudi::DataHandle::Writer, this};
-  DataHandle<int>                m_singleIntHandle{"SingleInt", Gaudi::DataHandle::Writer, this};
+  DataHandle<podio::UserDataCollection<float>> m_vectorFloatHandle{"VectorFloat", Gaudi::DataHandle::Writer, this};
```

Comment:
Using an `std::vector<float>` here would result in a compile time error, right? Otherwise we might have to make that the case to avoid runtime issues and add the necessary documentation to use the `UserDataCollection`

Reply:
at the moment it creates a runtime error.
in the end it is a matter of documentation that non-PODIO types will never be stored. We can give a warning somewhere in the process.